server/src/test/java/org/elasticsearch/index/query/MultiMatchQueryBuilderTests.java

```diff
@@ -240,6 +242,51 @@ public void testToQueryFieldMissing() throws Exception {
             instanceOf(MatchNoDocsQuery.class));
     }
 
+    public void testToQueryBooleanPrefixSingleField() throws IOException {
+        final MultiMatchQueryBuilder builder = new MultiMatchQueryBuilder("foo bar", STRING_FIELD_NAME);
+        builder.type(Type.BOOLEAN_PREFIX);
+        final Query query = builder.toQuery(createShardContext());
+        assertThat(query, instanceOf(BooleanQuery.class));
+        final BooleanQuery booleanQuery = (BooleanQuery) query;
+        assertThat(booleanQuery.clauses(), hasSize(2));
+        assertThat(assertBooleanSubQuery(booleanQuery, TermQuery.class, 0).getTerm(), equalTo(new Term(STRING_FIELD_NAME, "foo")));
+        assertThat(assertBooleanSubQuery(booleanQuery, PrefixQuery.class, 1).getPrefix(), equalTo(new Term(STRING_FIELD_NAME, "bar")));
+    }
+
+    public void testToQueryBooleanPrefixMultipleFields() throws IOException {
```

Comment:
@jimczi I think this behavior looks correct - with a text field and an alias field, we get a dismax with two boolean_prefix queries. With a text field and a keyword field, we get a dismax with a boolean_prefix on the text field and a prefix query on the keyword field

Reply:
I agree, this is the expected behavior